src/olympia/addons/models.py

```diff
@@ -1777,16 +1768,27 @@ def set_tag_list(self, new_tag_list):
         self.tag_list = new_tag_list
 
     def update_all_due_dates(self):
-        for version in self.versions.should_have_due_date().filter(
-            due_date__isnull=True
+        """
+        Update all due dates on versions of this add-on.
+        """
+        # This method should also affects deleted versions, so be careful when
+        # modifying.
```

Comment:
This was an attempt to deal with the fact that deleted unlisted versions should keep their due date if they have one. Andreas mentioned this in https://github.com/mozilla/addons-server/issues/20152#issuecomment-1397177487 (a Version signed but deleted should keep their due date)

I didn't finish this because it turned out to be more complex and deserved its own issue. I'll remove it, but we'll need something like that for the issue I linked to.

Reply:
https://github.com/mozilla/addons-server/pull/20181/commits/05816b97b56be76bb8a47c22aa9fe4fdae7cd183